### PR TITLE
DCS-967 Adding mechanism for developers to edit raw data of reports.

### PR DESCRIPTION
**Pull request by andrewrlee**
This is hidden behind the batch load role and is the first step towards working out how we actually allow users to do it

### CODE REVIEW

File: server/data/draftReportClient.ts [hunk 1]
```diff
@@ -32,22 +32,6 @@ export default class DraftReportClient {
...
-  ): Promise<void> {
-    await query({
-      text: `update v_report r
-            set form_response = COALESCE($1,   r.form_response)
-            ,   incident_date = COALESCE($2,   r.incident_date)
-            ,   updated_date = now()
-            where r.id = $3`,
-      values: [formResponse, incidentDate, reportId],
-    })
-  }
```
Comment (andrewrlee) [inline server/data/draftReportClient.ts:49]: this has been moved across to the incident client

File: server/data/incidentClientTypes.ts [hunk 2]
```diff
@@ -32,7 +33,7 @@ export interface Report extends ReportSummary {
   username: string
   agencyId: string
   submittedDate: Date
-  form: Record<string, any>
+  form: UseOfForceReport
```
Comment (andrewrlee) [inline server/data/incidentClientTypes.ts:36]: using more specific type

File: server/middleware/authorisationMiddleware.test.ts [hunk 3]
```diff
@@ -0,0 +1,158 @@
+import { Response } from 'express'
```
Comment (andrewrlee) [inline server/middleware/authorisationMiddleware.test.ts:1]: now typescripted

File: server/middleware/authorisationMiddleware.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import { NextFunction, Request, Response } from 'express'
+import jwtDecode from 'jwt-decode'
```
Comment (andrewrlee) [inline server/middleware/authorisationMiddleware.ts:2]: now typescripted

File: server/routes/maintainingReports/admin.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+    return res.render('pages/admin/edit-form.html', {
+      errors: req.flash('errors'),
+      data: {
+        incidentId: reportId,
+        reporterName,
+        submittedDate,
+        offenderDetail,
+        selectedFormName: formName,
+        formSections: Object.keys(form),
```
Comment (andrewrlee) [inline server/routes/maintainingReports/admin.ts:35]: should probably not use keys and instead order these appropriately and remove involved staff

**Review by GurnankCheema — APPROVED** [review]
LGTM

**Pull request merged**